server/app/controllers/applicant/ApplicantRequestedAction.java

```diff
@@ -19,4 +19,18 @@ public enum ApplicantRequestedAction {
   REVIEW_PAGE;
 
   public static final ApplicantRequestedAction DEFAULT_ACTION = NEXT_BLOCK;
+
+  /**
+   * Removes the applicant-requested action from the end of the provided {@code url} and returns the
+   * stripped URL, or returns the original URL if it doesn't end in an applicant-requested action.
+   */
+  public static String stripActionFromEndOfUrl(String url) {
+    for (ApplicantRequestedAction action : ApplicantRequestedAction.values()) {
+      String actionString = "/" + action.name();
+      if (url.endsWith(actionString)) {
+        return url.substring(0, url.indexOf(actionString));
```

Comment:
Would it be better to use the `Uri` standard library class to modify the URL, in case it happens to have a query string or other params on the end that we want to preserve?

Or will that never happen here?

Reply:
At the moment I don't think it can happen here but I think this is a good idea, updated.